### PR TITLE
Add download cache

### PR DESCRIPTION
**Pull request by TimoRoth**
Caches OGGM downloads to speed up build

### CODE REVIEW

File: .github/workflows/build-pages.yml [hunk 1]
```diff
@@ -34,7 +39,7 @@ jobs:
       - name: Push Book
         if: github.event_name != 'pull_request'
         run: |
-          ${PIP} install git+https://github.com/TimoRoth/ghp-import.git@master
+          ${PIP} install ghp-import
```
Comment (fmaussion) [inline .github/workflows/build-pages.yml:42]: What was the reason to use yours in the first place? I thought something about `stable` vs `latest` build?

Comment (TimoRoth) [reply .github/workflows/build-pages.yml:42]: The ability to have multiple pages in parallel without each wiping out the other.

Comment (fmaussion) [reply .github/workflows/build-pages.yml:42]: Ah right: https://github.com/c-w/ghp-import/pull/98

**Comment by fmaussion** [comment]
This seems ready to merge, what do you think @TimoRoth ?

**Comment by TimoRoth** [comment]
No, it doesn't work.
The big issue is that the Github-Cache is made for stuff like npm, which has a lock-file with discrete versions of its dependencies.
It will only update the cache when the cache-key changes, but the cache key has to be known in advance.
So basically I'd have to know exactly which files the build is going to download, before downloading them.
I'm close to concluding that it's simply impossible and my inquieries about this with Github Support have been pretty much "you're holding it wrong".

**Comment by TimoRoth** [comment]
I also thought "well, occasionally clearing the cache is probably good enough".
But: https://github.com/actions/cache/issues/2
It's impossible to clear it.

**Comment by TimoRoth** [comment]
It actually works now, with some patching around in the cache action.

**Pull request merged**